### PR TITLE
feat: support compute DER for multiple pairs of ref and hyp

### PR DESCRIPTION
**Pull request by LingweiMeng**
Support compute DER for multiple pairs of ref and hyp, by inputting `dict` or `list` type of ref and hyp.
Please check README and the code.

### CODE REVIEW

**Comment by desh2608** [comment]
Thanks for your contribution. I'll review this later, but please ping here if I don't get around to it by end of this month.

**Comment by LingweiMeng** [comment]
Hi @desh2608. I came across a problem with this commit.

The `double free or corruption (out)` error occurred at the below position, when I was computing multiple DERs during the training of a fairseq-based model. 
https://github.com/desh2608/spyder/blob/9dda8f40f2f2d54eade2a129d1394d00b7d1d1c5/src/spyder/der.cc#L137

Everything seems OK when out of the model training. 
Do you have any suggestions on this? Thanks so much for your help.

**Comment by desh2608** [comment]
Can you look at the segments for which the error occurred and see if you can find something weird? This may be a corner case that needs to be fixed.

**Comment by LingweiMeng** [comment]
> Can you look at the segments for which the error occurred and see if you can find something weird? This may be a corner case that needs to be fixed.

I'll check this later.

**Comment by LingweiMeng** [comment]
Hi @desh2608. I believe the reason is memory conflict when I use multi-thread training. 
Everything looks good when inference. This PR does not cause the error.

File: src/spyder/der.py [hunk 1]
```diff
@@ -4,15 +4,23 @@
...
-        self.conf = metrics.conf
-        self.der = metrics.der
+        try:
+            self.duration = metrics.duration
+            self.miss = metrics.miss
+            self.falarm = metrics.falarm
+            self.conf = metrics.conf
+            self.der = metrics.der
+        except:
+            self.duration = metrics["duration"]
```
Comment (desh2608) [inline src/spyder/der.py]: Why is this needed? Also, you should specify the exception type (bare except is not recommended as per pep8 rules).

**Review by desh2608 — CHANGES_REQUESTED** [review]
Small comment. Otherwise, LGTM.

File: src/spyder/der.py (hunk 1 above)
Comment (LingweiMeng) [reply src/spyder/der.py]: Because I used `dict` in line 90:
```
    return {m[0]: DERMetrics(dict(zip(['duration', 'miss', 'falarm', 'conf', 'der'], m[1:])))
            for m in selected_metrics}
```

Would you mind if I force-pushed this part as follows?
```
        if isinstance(metrics, dict):
            self.duration = metrics["duration"]
            self.miss = metrics["miss"]
            self.falarm = metrics["falarm"]
            self.conf = metrics["conf"]
            self.der = metrics["der"]
        else:
            self.duration = metrics.duration
            self.miss = metrics.miss
            self.falarm = metrics.falarm
            self.conf = metrics.conf
            self.der = metrics.der
```

Comment (desh2608) [reply src/spyder/der.py]: Is it necessary to return it as a dict? Could you just return it as an object of type `DERMetrics`?

Comment (LingweiMeng) [reply src/spyder/der.py]: I've pushed a new commit. Thanks for reviewing.

**Review by desh2608 — APPROVED** [review]

**Comment by desh2608** [comment]
LGTM. Thanks for your contribution!
